### PR TITLE
truncate trait keys when sending custom merge tags

### LINKED ISSUE #4: custom traits cant be longer than 10 chars
their merge fields, which we use to map custom traits, can't be longer than 10 chars.

this is a bummer for users who have been sending data to other tools with traits longer than 10 chars, as they're outright incompatible with mailchimp custom fields.

so we could do one of two things, (i think?):

- add a field mapper to the UI so that customers can map any trait (of any length) to specific merge fields.

- automatically take the first 10 character substring of any trait name longer than 10 chars and match on that basis instead 

of the two, i'm not sure which i dislike less. thoughts? @amillet89 @DirtyAnalytics @harrietgrace

### PR DESCRIPTION
**Pull request by sperand-io**
Closes #4. @amillet89 @f2prateek @ndhoule 

---

@daxhuiberts @zachkinloch @pauloschilling @DirtyAnalytics Sorry for the delay on this, folks!

### CODE REVIEW

File: lib/mapper.js [hunk 1]
```diff
@@ -23,7 +21,7 @@ exports.identify = function(identify){
 
   return extend(payload, {
     apikey: this.settings.apiKey,
-    id: find(opts, 'listId') || this.settings.listId,
+    id: object(opts, 'listId') || this.settings.listId,
```
Comment (f2prateek) [inline lib/mapper.js]: I'm not too familiar with this, can you clarify why this change was made?

**Comment by f2prateek** [comment]
LGTM, just one question.

File: lib/mapper.js (hunk 1 above)
Comment (pauloschilling) [inline lib/mapper.js]: I think it's just a code cleanup, since `find` was a instance of `obj-case` just like `object` is (lines 6-10).

Comment (sperand-io) [inline lib/mapper.js]: yep, exactly!

**Comment by sperand-io** [comment]
Thanks Prateek! I was just tidying code; both `find` and `object` were instances of `obj-case`, each with their own `require()` statement. See old version [here](https://github.com/segmentio/integration-mailchimp/pull/8/files#diff-0d856c89cb532015bc1efbae0bafe512L10). @f2prateek

**Comment by f2prateek** [comment]
sweet, lgtm!
